liberty-managed/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -1201,6 +1218,343 @@ private void throwDeploymentException(String applicationName) throws DeploymentE
+                   br.close();
+               } catch (IOException e) {
+                   log.warning(e.getMessage());
+               }
+           }
+       }
+       log.finest("Exception stack found was: " + Arrays.deepToString(result.toArray()) );
+       return result;
+   }
+
+   /**
+    * Find the app's StateChangeException FFDC file
+    * (using Java 6 compatible code).
+    * 
+    * @param appName
+    * @param ffdcFiles
+    * 
+    * @return The correct FFDC File or null 
+    * 
+    * @throws FileNotFoundException
+    * @throws IOException
+    */
+   private File findStateChangeExceptionFfdcFileForApp(String appName, File[] ffdcFiles) {
+
+       BufferedReader br = null;
+       try {
+           for (int i = 0; i < ffdcFiles.length; i++) {
+               
+               // Set up a bufferedReader
+               File ffdcFile = ffdcFiles[i];
+               log.finest("Processing FFDC file: " + ffdcFile.getAbsolutePath());
+               try {
+                   br = new BufferedReader(new InputStreamReader(new FileInputStream(ffdcFile)));
+               } catch (FileNotFoundException e) {
+                   log.warning("File " + ffdcFile.getAbsolutePath() + " is no longer accessible");
+                   continue; // File has been deleted but it might not have been the target one
+               }
+
+               // See if it fulfills the criteria
+               try {
+
+                   boolean fileIsForStateChangeException = false;
+                   boolean fileIsForCorrectApplication = false;
+
+                   String  line;
+                   while ((line = br.readLine()) != null) {
+                       
+                       fileIsForStateChangeException = fileIsForStateChangeException
+                               || line.contains("Stack Dump = com.ibm.ws.container.service.state.StateChangeException");
+                       log.finest("fileIsForStateChangeException:" + fileIsForStateChangeException + " " + line);
+                       
+                       fileIsForCorrectApplication = fileIsForCorrectApplication || line.contains("appName") && line.contains(appName);
+                       log.finest("fileIsForCorrectApplication:" + fileIsForCorrectApplication + "(" + appName + ")"  + " " + line);
+
+                       // Have we found all the criteria we are looking for?
+                       if (fileIsForStateChangeException && fileIsForCorrectApplication) {
+                           // We have found what we are looking for
+                           log.finest("Found StateChangeExceptionFfdc for " + appName + ": " + ffdcFile.getAbsolutePath());
+                           return ffdcFile;
+                       }
+                       
+                   }
+                   log.finest("Did not find StateChangeExceptionFfdc for " + appName + ": " + ffdcFile.getAbsolutePath() + "(file was scanned completely)");
+                   return ffdcFile;
```

Comment:
This looks like you'll always return the first ffdcFile even if it doesn't find a StateChangeException. I think this line should just be removed so that you continue round the loop.

Reply:
Done